### PR TITLE
Feature/add discord linking

### PR DESCRIPTION
**Pull request by AtrumAmantus**
Added ability to link a discord account
- Added endpoint
- Added properties configuration
- Added Models
- Updated property classes naming for clarity
- Added tests

### CODE REVIEW

**Comment by AtrumAmantus** [comment]
Not sure why previous 3 commits already in develop are showing in this one

**Comment by AtrumAmantus** [comment]
Get your discord code,
send code with bearer token in auth header to
*/authenticate/discord?code={code}
Check h2 db

File: src/main/java/dev/codesupport/web/common/security/AuthorizationService.java [hunk 1]
```diff
@@ -10,4 +11,7 @@
 
     UserDetails getUserDetailsByEmail(String email);
 
+    @PreAuthorize("hasPermission('discord', 'link_account')")
```
Comment (LamboCreeper) [inline src/main/java/dev/codesupport/web/common/security/AuthorizationService.java:14]: Everyone should have this permission by default.

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -9,12 +9,15 @@ security:
   jwt:
     issuer: ${jwt.issuer:CodeSupport.dev}
     expiration: ${jwt.expiration:10m}
+  discord:
```
Comment (LamboCreeper) [inline src/main/resources/application.yml:12]: Are all the values for the keys in `discord` meant to have colons at the end?

File: src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java [hunk 3]
```diff
@@ -6,67 +6,67 @@
...
         assertEquals(expected, actual);
     }
 
     @Test(expected = ConfigurationException.class)
     public void shouldThrowConfigurationExceptionForInvalidTimeFormat() {
-        JwtConfiguration jwtConfiguration = new JwtConfiguration();
+        JwtProperties jwtProperties = new JwtProperties();
 
-        jwtConfiguration.setExpiration("-2m");
+        jwtProperties.setExpiration("-2m");
```
Comment (LamboCreeper) [inline src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java:51]: Negative?

File: src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java [hunk 4]
```diff
@@ -6,67 +6,67 @@
...
     }
 
     @Test(expected = ConfigurationException.class)
     public void shouldThrowConfigurationExceptionForInvalidUnitFormat() {
-        JwtConfiguration jwtConfiguration = new JwtConfiguration();
+        JwtProperties jwtProperties = new JwtProperties();
 
-        jwtConfiguration.setExpiration("2r");
+        jwtProperties.setExpiration("2r");
```
Comment (LamboCreeper) [inline src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java:58]: What's `2r`?

**Review by LamboCreeper — COMMENTED** [review]
Looks good, just a couple of questions.

File: src/main/java/dev/codesupport/web/common/security/AuthorizationService.java (hunk 1 above)
Comment (AtrumAmantus) [reply src/main/java/dev/codesupport/web/common/security/AuthorizationService.java:14]: The only restriction this adds is that you are logged in, if you aren't, it doesn't know who to link the discordId to

File: src/main/resources/application.yml (hunk 2 above)
Comment (AtrumAmantus) [reply src/main/resources/application.yml:12]: If you mean the ones like `discord:` that's required yaml syntax, if you mean inside the ${} expression, that denotes default value

Comment (LamboCreeper) [reply src/main/resources/application.yml:12]: I meant before the `}`

File: src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java (hunk 3 above)
Comment (AtrumAmantus) [reply src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java:51]: shouldThrowConfigurationExceptionForInvalidTimeFormat()

File: src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java (hunk 4 above)
Comment (AtrumAmantus) [reply src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java:58]: shouldThrowConfigurationExceptionForInvalidUnitFormat()

File: src/main/resources/application.yml (hunk 2 above)
Comment (AtrumAmantus) [reply src/main/resources/application.yml:12]: Defaults it to a blank string, not necessary for logic, but makes a more clear error message later if not set

File: src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java (hunk 3 above)
Comment (AtrumAmantus) [reply src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java:51]: This is a unit test

File: src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java (hunk 4 above)
Comment (AtrumAmantus) [reply src/test/java/dev/codesupport/web/common/configuration/JwtPropertiesTest.java:58]: This is a unit test

**Review by LamboCreeper — APPROVED** [review]

**Pull request merged**